frappe/model/db_query.py

```diff
@@ -127,6 +127,22 @@ def build_and_run(self):
 		if self.distinct:
 			args.fields = 'distinct ' + args.fields
 			args.order_by = '' # TODO: recheck for alternative
+			
+		# Postgres requires any field that appears in the select clause to also 
+		# appear in the order by and group by clause
+		if args.order_by and args.group_by:
+			order_field = args.order_by
+
+			for r in (" order by ", " asc", " ASC", " desc", " DESC"):
+				order_field = order_field.replace(r, "")
+
+			if order_field not in args.fields:
+				order_fieldm = order_field.replace("`", "")
+				if "." in order_fieldm:
+					args.fields += ", MAX(" + order_fieldm.split(".")[1] + ") as `" + order_fieldm + "`"
+				else:
+					args.fields += ", MAX(" + order_fieldm + ") as `" + order_fieldm + "`"
+				args.order_by = args.order_by.replace(order_field, "`" + order_fieldm + "`")
```

Comment:
This should probably be moved in a separate method, perhaps under `prepare_select_args` and have this logic executed only if db_type is postgres...given it's unnecessary processing for MariaDB

Reply:
Done